compiler/rustc_codegen_ssa/src/mir/block.rs

```diff
@@ -103,33 +104,39 @@ impl<'a, 'tcx> TerminatorCodegenHelper<'tcx> {
     fn do_call<Bx: BuilderMethods<'a, 'tcx>>(
         &self,
         fx: &mut FunctionCx<'a, 'tcx, Bx>,
-        bx: &mut Bx,
+        mut bx: Bx,
         fn_abi: FnAbi<'tcx, Ty<'tcx>>,
         fn_ptr: Bx::Value,
         llargs: &[Bx::Value],
         destination: Option<(ReturnDest<'tcx, Bx::Value>, mir::BasicBlock)>,
         cleanup: Option<mir::BasicBlock>,
-    ) {
+    ) -> Bx::Unpositioned {
         // If there is a cleanup block and the function we're calling can unwind, then
         // do an invoke, otherwise do a call.
         if let Some(cleanup) = cleanup.filter(|_| fn_abi.can_unwind) {
-            let ret_bx = if let Some((_, target)) = destination {
+            let ret_llbb = if let Some((_, target)) = destination {
                 fx.blocks[target]
             } else {
                 fx.unreachable_block()
             };
-            let invokeret =
-                bx.invoke(fn_ptr, &llargs, ret_bx, self.llblock(fx, cleanup), self.funclet(fx));
-            bx.apply_attrs_callsite(&fn_abi, invokeret);
+            let (unpositioned_bx, invokeret) = bx.invoke(
+                fn_ptr,
+                &llargs,
+                ret_llbb,
+                self.llblock(fx, cleanup),
+                self.funclet(fx),
+                Some(&fn_abi),
+            );
 
             if let Some((ret_dest, target)) = destination {
                 let mut ret_bx = fx.build_block(target);
```

Comment:
It should be possible to re-use the `unpositioned_bx` here I think.

Reply:
Yupp! There's a lot of places like this, that I would want to tackle if this PR gets merged.